app/src/main/java/net/skyscanner/backpack/demo/compose/PageIndicatorStory.kt

```diff
@@ -0,0 +1,109 @@
+
+package net.skyscanner.backpack.demo.compose
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import net.skyscanner.backpack.compose.button.BpkButton
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicator
+import net.skyscanner.backpack.compose.pageindicator.BpkPageIndicatorStyle
+import net.skyscanner.backpack.compose.text.BpkText
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import net.skyscanner.backpack.demo.R
+
+@Composable
+@Preview
+fun PageIndicatorStory(
+  modifier: Modifier = Modifier,
+) {
+
+  Column(
+    modifier = modifier
+      .fillMaxSize()
+      .padding(BpkSpacing.Base),
+    horizontalAlignment = Alignment.CenterHorizontally,
+    verticalArrangement = Arrangement.spacedBy(BpkSpacing.Base),
+  ) {
+    BpkText(text = stringResource(id = R.string.page_indicator_less_than_5))
+    PageIndicatorSample(totalIndicators = 3, style = BpkPageIndicatorStyle.Default)
+
+    BpkText(text = stringResource(id = R.string.page_indicator_more_than_5))
+    PageIndicatorSample(totalIndicators = 8, style = BpkPageIndicatorStyle.Default)
+
+    BpkText(text = stringResource(id = R.string.page_indicator_over_image))
+    Box(
+      modifier = Modifier
+        .fillMaxWidth()
+        .height(BpkSpacing.Xxl * 2),
+      contentAlignment = Alignment.BottomCenter,
+    ) {
+      Image(
+        painter = painterResource(id = R.drawable.canadian_rockies_canada),
+        contentDescription = "",
+        contentScale = ContentScale.FillWidth,
+      )
+      PageIndicatorSample(totalIndicators = 8, style = BpkPageIndicatorStyle.OverImage)
+    }
+  }
+}
+
+@Composable
+private fun PageIndicatorSample(
+  totalIndicators: Int,
+  style: BpkPageIndicatorStyle,
+  modifier: Modifier = Modifier,
+) {
+  var currentIndex by remember { mutableStateOf(0) }
+
+  Row(
+    modifier = modifier.fillMaxWidth(),
+    horizontalArrangement = Arrangement.SpaceEvenly,
+    verticalAlignment = Alignment.CenterVertically,
+  ) {
+    BpkButton(text = stringResource(id = R.string.page_indicator_prev)) {
+      currentIndex = maxOf(currentIndex - 1, 0)
```

Comment:
I think we should coerce values inside PageIndicator itself before passing to lambda